src/ReverseIterator.ts

```diff
@@ -0,0 +1,188 @@
+import { compare, Time, subtract as subTime } from "@foxglove/rostime";
+import Heap from "heap";
+
+import { IBagReader } from "./IBagReader";
+import { ChunkInfo, MessageData } from "./record";
+import { IteratorConstructorArgs, Decompress, ChunkReadResult } from "./types";
+
+class ReverseIterator {
```

Comment:
```suggestion
export class ReverseIterator {
```

Reply:
These two classes look very similar, is there a way we can mostly reuse all the code except flipping some conditionals?

Reply:
Possibly - At least the generator itself is the same and could come from a base class I suppose.